tests/parser/types/value/test_as_wei_value.py

```diff
@@ -1,30 +1,84 @@
-def test_wei_conversion(get_contract_with_gas_estimation):
-    test_wei = """
-@external
-def return_2_finney() -> uint256:
-    return as_wei_value(2, "finney")
+from decimal import Decimal
+
+import pytest
+
+wei_denoms = {
+    ("femtoether", "kwei", "babbage"): 3,
+    ("picoether", "mwei", "lovelace"): 6,
+    ("nanoether", "gwei", "shannon"): 9,
+    ("microether", "szabo",): 12,
+    ("milliether", "finney",): 15,
+    ("ether",): 18,
+    ("kether", "grand"): 21,
+}
```

Comment:
It would be clearer if this were in a flat heirarchy
```suggestion
wei_denoms = {
    "femtoether": 3,
    "kwei": 3,
    "babbage": 3,
    "picoether": 6,
    "mwei": 6,
    "lovelace": 6,
    "nanoether": 9,
    "gwei": 9,
    "shannon": 9,
    "microether": 12,
    "szabo": 12,
    "milliether": 15,
    "finney": 15,
    "ether": 18,
    "kether": 21,
    "grand": 21,
}
```